crates/ra_hir_ty/src/display.rs

```diff
@@ -321,8 +319,8 @@ impl HirDisplay for Ty {
     }
 }
 
-fn write_bounds_like_dyn_trait(
-    predicates: &[GenericPredicate],
+fn write_bounds_like_dyn_trait<'a>(
+    predicates: impl IntoIterator<Item = impl Borrow<GenericPredicate>>,
```

Comment:
That feel, when one invocation passes `Item = GenericPredicate`, but the other is `Item = &GenericPredicate`. Not sure if my solution is overengineered, I can leave intermediate collect::<Vec<_>> if you wish